pkg/applicationserver/io/packages/alcsync/v1/package.go

```diff
@@ -0,0 +1,109 @@
+// Copyright © 2023 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package alcsyncv1
+
+import (
+	"context"
+
+	"go.thethings.network/lorawan-stack/v3/pkg/applicationserver/io"
+	"go.thethings.network/lorawan-stack/v3/pkg/applicationserver/io/packages"
+	"go.thethings.network/lorawan-stack/v3/pkg/errors"
+	"go.thethings.network/lorawan-stack/v3/pkg/log"
+	"go.thethings.network/lorawan-stack/v3/pkg/ttnpb"
+)
+
+// PackageName is the name of the package.
+const PackageName = "alcsync-v1"
+
+var errNoAssociation = errors.DefineInternal("no_association", "no association available")
+
+type alcsyncpkg struct {
+	server     io.Server
+	registry   packages.Registry
+	cmdFactory CommandFactory
+}
+
+// HandleUp implements packages.ApplicationPackageHandler.
+func (a *alcsyncpkg) HandleUp(
+	ctx context.Context,
+	def *ttnpb.ApplicationPackageDefaultAssociation,
+	assoc *ttnpb.ApplicationPackageAssociation,
+	up *ttnpb.ApplicationUp,
+) error {
+	ctx = log.NewContextWithField(ctx, "namespace", "applicationserver/io/packages/alcsync/v1")
+	logger := log.FromContext(ctx)
+
+	logger.Info("Handle uplink")
+
+	if def == nil && assoc == nil {
+		logger.Info("No association available")
+		return errNoAssociation.New()
+	}
+
+	msg, ok := up.Up.(*ttnpb.ApplicationUp_UplinkMessage)
+	if !ok {
+		logger.Debug("Uplink is not an uplink message; drop payload")
```

Comment:
We don't really use `;` inside log messages. Use `.` as a sentence connector instead.

Also same below.